File: Backend/src/main/java/com/back/domain/vocabulary/entity/Vocabulary.java
package com.back.domain.vocabulary.entity;

import com.back.domain.user.entity.User;
import com.back.global.jpa.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
@Table(name = "vocabularies")
public class Vocabulary extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false, length = 100)
    private String word;

    @Column(nullable = false, length = 100)
    private String hiragana;

    @Column(nullable = false, length = 500)
    private String meaning;

    @Column(columnDefinition = "TEXT")
    private String exampleSentence;

    @Column(columnDefinition = "TEXT")
    private String exampleTranslation;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private StudyStatus studyStatus = StudyStatus.NOT_STUDIED;


    // ===== Helper Method ===== //
    public void update(String word, String hiragana, String meaning,
                       String exampleSentence, String exampleTranslation,
                       StudyStatus studyStatus) {
        this.word = word;
        this.hiragana = hiragana;
        this.meaning = meaning;
        this.exampleSentence = exampleSentence;
        this.exampleTranslation = exampleTranslation;
        this.studyStatus = studyStatus;
    }

    public void updateStudyStatus(StudyStatus studyStatus) {
        this.studyStatus = studyStatus;
    }
}

File: Frontend/types/vocabulary.ts
export type StudyStatus = 'NOT_STUDIED' | 'STUDYING' | 'COMPLETED';

export interface VocabularyResponse {
  id: number;
  word: string;
  hiragana: string;
  meaning: string;
  exampleSentence: string | null;
  exampleTranslation: string | null;
  studyStatus: StudyStatus;
  studyStatusDisplay: string;
  createDate: string;
  updateDate: string;
}

export interface VocabularyListResponse {
  id: number;
  word: string;
  hiragana: string;
  meaning: string;
  studyStatus: StudyStatus;
  studyStatusDisplay: string;
  createDate: string;
}

export interface VocabularyCreateRequest {
  word: string;
  hiragana: string;
  meaning: string;
  exampleSentence?: string;
  exampleTranslation?: string;
}

export interface VocabularyUpdateRequest {
  word: string;
  hiragana: string;
  meaning: string;
  exampleSentence?: string;
  exampleTranslation?: string;
  studyStatus: StudyStatus;
}